services/QuillLMS/lib/rule_feedback_history.rb

```diff
@@ -26,6 +26,31 @@ def self.exec_query(conjunction:, activity_id:)
         rules = Comprehension::Rule.find_by_sql(sql)
     end
 
+    def self.feedback_history_to_json(f_h)
```

Comment:
This is sooooo much easier to read than the previous version of this method. Nice job.